src/callback_handler/oci.rs

```diff
@@ -15,9 +16,16 @@ impl Client {
 
     /// Fetch the manifest digest of the OCI resource referenced via `image`
     pub async fn digest(&self, image: &str) -> Result<String> {
-        let image_with_proto = format!("registry://{}", image);
-        self.registry
+        // this is needed to expand names as `busybox` into
+        // fully resolved references like `docker.io/library/busybox`
+        let image_ref: Reference = image.parse()?;
+
+        let image_with_proto = format!("registry://{}", image_ref.whole());
+        let image_digest = self
+            .registry
             .manifest_digest(&image_with_proto, self.sources.as_ref())
-            .await
+            .await?;
+        serde_json::to_string(&image_digest)
+            .map_err(|e| anyhow!("Cannot serialize response to json: {}", e))
```

Comment:
why do we need this? `manifest_digest` returns a `Result<String>`. Can we just return `manifest_digest` ?

Reply:
I am doubting myself too here.
This change is consistent with the rest of callbacks, though. But in https://github.com/kubewarden/policy-evaluator/blob/main/src/callback_handler/mod.rs#L153, we use the digest as a vector of bytes, instead of doing a `serde_json::to_vec()`.

Maybe I'm misunderstanding things, but, for `CallbackRequestType::SigstoreKeylessVerify` and `CallbackRequestType::SigstorePubKeysVerify`, don't we transform the payload of the `CallbackResponse` to json twice? (so we create a json from a json)

Reply:
ah actually I think we need it. `manifest_digest` returns a string, and `serde_json::to_string` will return a valid JSON for this String.
Then https://github.com/kubewarden/policy-evaluator/blob/main/src/callback_handler/mod.rs#L153 we just transform it to Vec<u8>, then we get the String back in the sdk.

Reply:
> Maybe I'm misunderstanding things, but, for CallbackRequestType::SigstoreKeylessVerify and CallbackRequestType::SigstorePubKeysVerify, don't we transform the payload of the CallbackResponse to json twice? (so we create a json from a json). In the past I though that it was because we want to always return a json object, that wraps the response. Hence why we did it here for the digest. Am I understanding it correctly?

I think we just transform it to JSON once we are about to send it. [Here](https://github.com/kubewarden/policy-evaluator/blob/main/src/callback_handler/sigstore_verification.rs#L56) we just return a `VerificationResponse`, then we convert it to json [here](https://github.com/kubewarden/policy-evaluator/blob/main/src/callback_handler/mod.rs#L195)

Reply:
Resolving if you agree @viccuad? since I now think this is fine, sorry about the noise

Reply:
Yes, I see it now. Thanks!